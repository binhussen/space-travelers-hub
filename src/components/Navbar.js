/** @format */

import { NavLink } from 'react-router-dom';
import logo from '../assets/logo.png';

const Navbar = () => (
  <nav className="w-screen bg-white p-2 flex items-center justify-around">
    <div className="flex items-center gap-8">
      <img className="w-16" src={logo} alt="logo" />
      <h1 className="font-bold">Space Travelers&apos; Hub</h1>
    </div>
    <ul className="flex items-center list-none gap-8">
      <li>
        <NavLink
          className="text-black text-lg no-underline hover:text-sky-500 hover:underline font-normal"
          to="/"
        >
          Rockets
        </NavLink>
      </li>
      <li>
        <NavLink
          className="text-black text-lg no-underline hover:text-sky-500 hover:underline font-normal"
          to="/missions"
        >
          Missions
        </NavLink>
      </li>
      <li>
        <NavLink
          className="text-black text-lg no-underline hover:text-sky-500 hover:underline font-normal"
          to="/profile"
        >
          My Profile
        </NavLink>
      </li>
    </ul>
  </nav>
);

export default Navbar;
