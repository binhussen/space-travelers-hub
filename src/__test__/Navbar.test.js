/** @format */

import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../components/Navbar';

describe('Navbar Component', () => {
  test('Builds the snapshot of the Navbar component', () => {
    const navbar = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    expect(navbar).toMatchSnapshot();
  });
});
