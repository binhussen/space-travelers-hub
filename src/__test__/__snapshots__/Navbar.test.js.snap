// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component Builds the snapshot of the Navbar component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="w-screen bg-white p-2 flex items-center justify-around"
      >
        <div
          class="flex items-center gap-8"
        >
          <img
            alt="logo"
            class="w-16"
            src="logo.png"
          />
          <h1
            class="font-bold"
          >
            Space Travelers' Hub
          </h1>
        </div>
        <ul
          class="flex items-center list-none gap-8"
        >
          <li>
            <a
              aria-current="page"
              class="text-lg hover:text-sky-500 hover:underline active"
              href="/"
            >
              Rockets
            </a>
          </li>
          <li>
            <a
              class="text-lg hover:text-sky-500 hover:underline"
              href="/missions"
            >
              Missions
            </a>
          </li>
          <li>
            <a
              class="text-lg hover:text-sky-500 hover:underline"
              href="/profile"
            >
              My Profile
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="w-screen bg-white p-2 flex items-center justify-around"
    >
      <div
        class="flex items-center gap-8"
      >
        <img
          alt="logo"
          class="w-16"
          src="logo.png"
        />
        <h1
          class="font-bold"
        >
          Space Travelers' Hub
        </h1>
      </div>
      <ul
        class="flex items-center list-none gap-8"
      >
        <li>
          <a
            aria-current="page"
            class="text-lg hover:text-sky-500 hover:underline active"
            href="/"
          >
            Rockets
          </a>
        </li>
        <li>
          <a
            class="text-lg hover:text-sky-500 hover:underline"
            href="/missions"
          >
            Missions
          </a>
        </li>
        <li>
          <a
            class="text-lg hover:text-sky-500 hover:underline"
            href="/profile"
          >
            My Profile
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
